### ISSUE TITLE
2.0.0 In-Synth patches not capturing names

### ISSUE THREAD

**Issue by Mostelin** [christofmuc/KnobKraft-orm#217]
I am uncertain whether this is a bug, an oversight or a feature.

Activating the 'In synth' internal patches generally worked as expected, in that the correct number of patches (40) appear onscreen and they serve as convenient ways to send program changes to the synth when clicked. I like this bit.

The bit I am not so taken with is that the patch names are replaced with a set of codes: 00-00, 00-01, 00-02, etc. I think these are made up of the layer and program indices (though the ESQ-1 does not have layers). Wouldn't it make more sense to show the user the patch name?

**Comment by christofmuc** [comment]
Hm, that must be a problem with the adaptations, because all synths I tested worked. But then, those are the old ones written in C++. I need to find a way to test this, will check!

**Comment by jontrulson** [comment]
I may have seen a similar problem.  I have a Korg MS2000, and I built this software about 2-3 weeks ago.  I noticed the same things when I loaded full sysex's - no patch names.  I wasn't sure if this was an issue with the adaptation (maybe being incomplete?) or what.

**Comment by christofmuc** [comment]
Hm, can you give me the MIDI log of one of these conversations? It seems both Ensoniq as well as the Korg MS2000 have the bank dump feature implemented - I have no synth that has bank dump *and* stored patch names. My Kawai K3 has bank dump, but no stored patch names, and the Sequentials do store patch names, but don't do bank dumps.

I will look into the code and see if I can spot something obvious,

**Comment by christofmuc** [comment]
I tested this again with a Sequential adaptation, and that works as expected as well. The logic is that the bank/number pair you are seeing should only be generated when the return value from nameFromDump() is empty.

For the Ensoniq: Can you see the print statement from the bank dump parsing in the log which shoudl read 
"Found patch x
Found patch y
Found patch z"

etc, which would indicate the bank dump works and the names also have worked.

Again, log files (MIDI log and regular log) would help!

**Comment by Mostelin** [comment]
I've attached both logs. From the MIDI log it looks like the Ensoniq names are being recognised.
[MIDI log.txt](https://github.com/christofmuc/KnobKraft-orm/files/10738910/MIDI.log.txt)
[Main log.txt](https://github.com/christofmuc/KnobKraft-orm/files/10738911/Main.log.txt)

**Comment by christofmuc** [comment]
Hm, that looks normal. What strikes me odd - all those patches are already known to the database, therefore none of them has really be imported. All that happens is that the synth bank now is a special list which points to the already existing patches. No names involved.

Two diagnostic ideas:
  - Try this in a fresh empty database so that the patches are really imported
  - Double check (sorry for that) that you have not set the display combo box to Program #, because that would actually cause exactly that behavior (display only though).

**Comment by Mostelin** [comment]
I tried both your suggestions using the new version 2.0.1 (thanks for all the fixes).

With an empty database:
- as before, codes of the form 00-00, 00-01, etc displayed in the main area, current patch tab, and synth bank tab.
- when selecting a program, main log shows as follows: 
10:50:09: info Sending program change to Ensoniq ESQ-1: program 00-03 [known bank]. 
10:50:14: info Sending program change to Ensoniq ESQ-1: program 00-02 [known bank]. 
10:50:15: info Sending program change to Ensoniq ESQ-1: program 00-01 [known bank]. 
10:50:16: info Sending program change to Ensoniq ESQ-1: program 00-00 [known bank]. 

Confirmed display combo box set to 'Name'.

both logs attached.

**New problem**: the first 13 patches are sometimes shown with numbers repeated bottom left, even when 'Name' selected in display combo box (see attached screen grab).

To recreate, select the bank entry under 'By Import', then reselect the bank entry under 'In Synth'.
To reset, select another option in display combo box, then reselect 'Name'.

[Main log 2.txt](https://github.com/christofmuc/KnobKraft-orm/files/10750690/Main.log.2.txt)
[MIDI log 2.txt](https://github.com/christofmuc/KnobKraft-orm/files/10750691/MIDI.log.2.txt)
![Patch number display bug](https://user-images.githubusercontent.com/95011279/219225562-fb6d35c1-b0a8-4b7d-bfc0-ffdaab6ecbc8.PNG)

**Comment by christofmuc** [comment]
@Mostelin @jontrulson I found this bug - the function 'renamePatch' accidentally was not optional. So when it came to loading the data, it would check if the storedPatchNameCapability was implemented by the adaptation, and because renamePatch was missing, it did not even try to extract the name. Changed and will be part of next release.

Note that if you do not implement renamePatch, the name in the database will be changeable and displayed in the Orm, but when you send it to synth the synth will still display the old name because the patch bytes haven't been changed.
